### PR TITLE
Okx: fetchLedger unify marginMode

### PR DESCRIPTION
**Pull request by Dan-krm**
Added marginMode to fetchLedger:

### Cross:
```
node examples/js/cli okx fetchLedger undefined undefined undefined '{"marginMode":"cross"}'

okx.fetchLedger (, , , [object Object])
2022-07-08T06:22:36.364Z iteration 0 passed in 377 ms

                id |     timestamp |                 datetime | account |        referenceId | referenceAccount |  type | currency |   symbol |  amount | before |              after | status |                                      fee
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
464807812252741636 | 1657090818152 | 2022-07-06T07:00:18.152Z |         | 464807812227575826 |                  | trade |     USDT | ETH/USDT |       0 |        |  40.92288696303541 |     ok |             {"cost":0,"currency":"USDT"}
464807812252741635 | 1657090818152 | 2022-07-06T07:00:18.152Z |         | 464807812227575826 |                  | trade |      ETH | ETH/USDT |       0 |        |        0.000908031 |     ok |    {"cost":0.000004583,"currency":"ETH"}
464807894226219048 | 1657090837696 | 2022-07-06T07:00:37.696Z |         |                    |                  | trade |      ETH | ETH/USDT |       0 |        |        0.000908031 |     ok |              {"cost":0,"currency":"ETH"}
...
20 objects
2022-07-08T06:22:36.364Z iteration 1 passed in 377 ms
```

### Isolated:
```
node examples/js/cli okx fetchLedger undefined undefined undefined '{"marginMode":"isolated"}'

okx.fetchLedger (, , , [object Object])
2022-07-08T06:23:25.253Z iteration 0 passed in 367 ms

                id |     timestamp |                 datetime | account |        referenceId | referenceAccount |  type | currency |   symbol |          amount | before |              after | status |                                       fee
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
464803126858657828 | 1657089701067 | 2022-07-06T06:41:41.067Z |         | 464803126829297665 |                  | trade |     USDT | ETH/USDT |               0 |        |  40.89741100923541 |     ok |              {"cost":0,"currency":"USDT"}
464803126858657827 | 1657089701067 | 2022-07-06T06:41:41.067Z |         | 464803126829297665 |                  | trade |      ETH | ETH/USDT |      -0.0001586 |        |         0.00080773 |     ok |         {"cost":7.93e-7,"currency":"ETH"}
464803126862852101 | 1657089701068 | 2022-07-06T06:41:41.068Z |         | 464803126829297665 |                  | trade |     USDT | ETH/USDT |               0 |        |  40.89741100923541 |     ok |              {"cost":0,"currency":"USDT"}
...
40 objects
2022-07-08T06:23:25.253Z iteration 1 passed in 367 ms
```

### CODE REVIEW

File: js/okx.js [hunk 1]
```diff
@@ -3172,6 +3175,11 @@ module.exports = class okx extends Exchange {
             // 'before': 'id', // return records newer than the requested bill id
             // 'limit': 100, // default 100, max 100
         };
+        const marginMode = this.safeString2 (params, 'mgnMode', 'marginMode');
```
Comment (carlosmiei) [inline js/okx.js:3178]: @Dan-krm can you update this with `handelMarginAndParam` please?

File: js/okx.js [hunk 2]
```diff
@@ -3173,6 +3176,11 @@ module.exports = class okx extends Exchange {
             // 'before': 'id', // return records newer than the requested bill id
             // 'limit': 100, // default 100, max 100
         };
+        const [ marginMode, marginModeQuery ] = this.handleMarginModeAndParams ('fetchLedger', params);
+        if (marginMode !== undefined) {
```
Comment (carlosmiei) [inline js/okx.js:3180]: @Dan-krm can we replicate the margin-logic applied in fetchLeverage here, for the sake of consistency?

File: js/okx.js [hunk 3]
```diff
@@ -3101,6 +3101,9 @@ module.exports = class okx extends Exchange {
          * @method
          * @name okx#fetchLedger
          * @description fetch the history of changes, actions done by the user or operations that altered balance of the user
+         * @see https://www.okx.com/docs-v5/en/#rest-api-account-get-bills-details-last-7-days
+         * @see https://www.okx.com/docs-v5/en/#rest-api-account-get-bills-details-last-3-months
+         * @see https://www.okx.com/docs-v5/en/#rest-api-funding-asset-bills-details
          * @param {string|undefined} code unified currency code, default is undefined
          * @param {int|undefined} since timestamp in ms of the earliest ledger entry, default is undefined
          * @param {int|undefined} limit max number of ledger entrys to return, default is undefined
```
Comment (carlosmiei) [inline js/okx.js:3109]: can we add params.marginMode here pls?

File: js/okx.js (hunk 2 above)
Comment (Dan-krm) [reply js/okx.js]: Okay I replicated the logic except I didn't default to cross margin since the mgnMode parameter is optional here.

Comment (carlosmiei) [reply js/okx.js]: @Dan-krm you did well, tkx!

**Comment by carlosmiei** [comment]
LGTM ✅ 

```

(base) ➜  ccxt git:(okxFetchLedger) ✗ p okx fetchLedger None None None '{"marginMode":"cross"}'   
Python v3.9.7
CCXT v1.91.100
okx.fetchLedger(None,None,None,{'marginMode': 'cross'})
[{'account': None,
  'after': 2.996948110976848,
  'amount': 5.4076869e-05,
  'before': None,
  'currency': 'USDT',
  'datetime': '2022-08-18T11:30:27.133Z',
  'fee': {'cost': 5.4131e-08, 'currency': 'USDT'},
  'id': '480458475865288808',
  'info': {'bal': '2.9969481109768481',
           'balChg': '0.0000540768690000',
           'billId': '480458475865288808',
           'ccy': 'USDT',
           'execType': 'T',
           'fee': '-0.000000054131',
           'from': '',
           'instId': 'ADA-USDT',
           'instType': 'SPOT',
           'mgnMode': 'cross',
           'notes': '',
           'ordId': '480458475835928579',
           'pnl': '0',
           'posBal': '0',
           'posBalChg': '0',
           'subType': '1',
           'sz': '0.000054131',
           'to': '',
           'ts': '1660822227133',
           'type': '2'},
  'referenceAccount': None,
  'referenceId': '480458475835928579',
  'status': 'ok',
  'symbol': 'ADA/USDT',
  'timestamp': 1660822227133,
  'type': 'trade'},
 {'account': None,
  'after': 17.60622812,
  'amount': -0.0001,
  'before': None,
  'currency': 'ADA',
  'datetime': '2022-08-18T11:30:27.133Z',
  'fee': {'cost': 0.0, 'currency': 'ADA'},
  'id': '480458475865288807',
  'info': {'bal': '17.60622812',
           'balChg': '-0.00010000',
           'billId': '480458475865288807',
           'ccy': 'ADA',
           'execType': 'T',
           'fee': '0',
           'from': '',
           'instId': 'ADA-USDT',
           'instType': 'SPOT',
           'mgnMode': 'cross',
           'notes': '',
           'ordId': '480458475835928579',
           'pnl': '0',
           'posBal': '0',
           'posBalChg': '0',
           'subType': '2',
           'sz': '0.0001',
           'to': '',
           'ts': '1660822227133',
           'type': '2'},
  'referenceAccount': None,
  'referenceId': '480458475835928579',
  'status': 'ok',
  'symbol': 'ADA/USDT',
  'timestamp': 1660822227133,
  'type': 'trade'}]
  
 ```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**